File: app/src/main/java/com/hazelmobile/filetransfer/ui/adapter/MusicListAdapter.java
package com.hazelmobile.filetransfer.ui.adapter;

import android.content.ContentResolver;
import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.provider.MediaStore;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.hazelmobile.filetransfer.R;
import com.hazelmobile.filetransfer.util.FileUtils;
import com.hazelmobile.filetransfer.app.EditableListAdapter;
import com.hazelmobile.filetransfer.object.Shareable;
import com.hazelmobile.filetransfer.util.TimeUtils;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class MusicListAdapter
        extends EditableListAdapter<MusicListAdapter.SongHolder, EditableListAdapter.EditableViewHolder> {

    private ContentResolver mResolver;

    public MusicListAdapter(Context context) {
        super(context);
        mResolver = context.getContentResolver();
    }

    @Override
    public List<MusicListAdapter.SongHolder> onLoad() {

        List<SongHolder> audioList = new ArrayList<>();

        Cursor songCursor = mResolver.query(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI,
                null,
                MediaStore.Audio.Media.IS_MUSIC + "=?",
                new String[]{String.valueOf(1)},
                null);

        if (songCursor != null) {
            if (songCursor.moveToFirst()) {

                int idIndex = songCursor.getColumnIndex(MediaStore.Audio.Media._ID);
                int artistIndex = songCursor.getColumnIndex(MediaStore.Audio.Media.ARTIST);
                int songIndex = songCursor.getColumnIndex(MediaStore.Audio.Media.TITLE);
                int nameIndex = songCursor.getColumnIndex(MediaStore.Audio.Media.DISPLAY_NAME);
                int dateIndex = songCursor.getColumnIndex(MediaStore.Audio.Media.DATE_MODIFIED);
                int sizeIndex = songCursor.getColumnIndex(MediaStore.Audio.Media.SIZE);
                int typeIndex = songCursor.getColumnIndex(MediaStore.Audio.Media.MIME_TYPE);

                do {
                    audioList.add(new SongHolder(
                            songCursor.getLong(idIndex),
                            songCursor.getString(nameIndex),
                            songCursor.getString(artistIndex),
                            songCursor.getString(songIndex),
                            songCursor.getString(typeIndex),
                            songCursor.getLong(dateIndex) * 1000,
                            songCursor.getLong(sizeIndex),
                            Uri.parse(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI + "/" + songCursor.getInt(idIndex)),
                            FileUtils.sizeExpression(songCursor.getLong(sizeIndex), false) + ", " +
                                    TimeUtils.INSTANCE.formatDateTime(getContext(), songCursor.getLong(dateIndex) * 1000)));
                }
                while (songCursor.moveToNext());
            }

            songCursor.close();
        }

        Collections.sort(audioList, getDefaultComparator());

        return audioList;
    }

    @NonNull
    @Override
    public EditableViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new EditableViewHolder(getInflater().inflate(R.layout.music_item_layout, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull EditableViewHolder holder, int position) {
        try {

            final SongHolder object = getItem(position);
            final View parentView = holder.getView();

            TextView text1 = parentView.findViewById(R.id.audioTitle);
            TextView text2 = parentView.findViewById(R.id.audioSize);
            text1.setText(object.song);
            text2.setText(object.datesize);

            parentView.setSelected(object.isSelectableSelected());

        } catch (Exception e) {

        }
    }

    public static class SongHolder extends Shareable {

        public String datesize;
        public String song;

        public SongHolder(long id, String displayName, String artist, String song, String mimeType, long date, long size, Uri uri, String datesize) {
            super(id, song + " - " + artist, displayName, mimeType, date, size, uri);
            this.datesize = datesize;
            this.song = song;
        }
    }


}

File: app/src/main/java/com/hazelmobile/filetransfer/ui/adapter/MusicListAdapter.kt
package com.hazelmobile.filetransfer.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.hazelmobile.filetransfer.R
import com.hazelmobile.filetransfer.model.SongHolder

class MusicListAdapter(private val musicItem: List<SongHolder>, val context: Context) :
    RecyclerView.Adapter<MusicListAdapter.MusicViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MusicViewHolder {
        return MusicViewHolder(
            LayoutInflater.from(context).inflate(
                R.layout.music_item_layout,
                parent,
                false
            )
        )
    }

    override fun getItemCount(): Int {
        return musicItem.size
    }

    override fun onBindViewHolder(holder: MusicViewHolder, position: Int) {
        val songHolder: SongHolder = musicItem[position]
        holder.bind(songHolder)
    }

    inner class MusicViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private var musicName: TextView? = null
        private var musicDateSize: TextView? = null

        init {
            musicName = itemView.findViewById(R.id.audioTitle)
            musicDateSize = itemView.findViewById(R.id.audioSize)
        }

        fun bind(songHolder: SongHolder) {
            musicName?.text = songHolder.displayName
            musicDateSize?.text = songHolder.datesize
        }
    }
}